packages/tool-cache/src/tool-cache.ts

```diff
@@ -83,23 +83,19 @@ export async function downloadTool(
 
       const file: NodeJS.WritableStream = fs.createWriteStream(dest)
       file.on('open', async () => {
-        try {
-          const stream = response.message.pipe(file)
-          stream.on('close', () => {
```

Comment:
`stream` is the writable stream. To improve readability - i had trouble figuring out which stream this is until after reading nodejs docs a bit - i moved the closed event handler further below (`file.on('close', ...)`)

Reply:
Which also makes it consistent with the error handler for the writable stream - registered on local variable `file` not `stream`